packages/alfa-style/src/node/predicate/is-scrolled-behind.ts

```diff
@@ -0,0 +1,166 @@
+    | Longhands.Specified<"overflow-y">,
+): boolean => overflow.value === "hidden";
+
+function isScrollContainerFor(
+  elementBox: Rectangle,
+  device: Device,
+): Predicate<Element> {
+  return function isScrollContainerFor(ancestor): boolean {
+    return or(
+      and(
+        // Element is in the scroll port, so it's not completely scrolled behind
+        not(
+          hasBox((ancestorBox) => ancestorBox.intersects(elementBox), device),
+        ),
+        or(
+          // Element intersects the rectangle to the right stretching to infinity
+          //
+          //    +-------+-------- - -
+          //    |       |  *
+          //    |       |
+          //    +-------+-------- - -
+          //
+          and(
+            hasBox(
+              (ancestorBox) =>
+                Rectangle.of(
+                  ancestorBox.right,
+                  ancestorBox.top,
+                  Infinity,
+                  ancestorBox.height,
+                ).intersects(elementBox),
+              device,
+            ),
+            hasComputedStyle("overflow-x", isScrollOrAuto, device),
+            hasComputedStyle(
+              "overflow-y",
+              or(isScrollOrAuto, isHidden),
+              device,
+            ),
+          ),
+          // Element intersects the rectangle below stretching to infinity
+          //
+          //    +-------+
+          //    |       |
+          //    |       |
+          //    +-------+
+          //    |   *   |
+          //    |       |
+          //    .       .
+          //    .       .
+          and(
+            hasBox(
+              (ancestorBox) =>
+                Rectangle.of(
+                  ancestorBox.left,
+                  ancestorBox.bottom,
+                  ancestorBox.width,
+                  Infinity,
+                ).intersects(elementBox),
+              device,
+            ),
+            hasComputedStyle(
+              "overflow-x",
+              or(isScrollOrAuto, isHidden),
+              device,
+            ),
+            hasComputedStyle("overflow-y", isScrollOrAuto, device),
+          ),
+          // Element intersects the rectangle to the right and below stretching to infinity
+          //
+          //    +-------+
+          //    |       |
+          //    |       |
+          //    +-------+-------- - -
+          //            |
+          //            |   *
+          //            .
+          //            .
+          and(
+            hasBox(
+              (ancestorBox) =>
+                Rectangle.of(
+                  ancestorBox.right,
+                  ancestorBox.bottom,
+                  Infinity,
+                  Infinity,
+                ).intersects(elementBox),
+              device,
+            ),
+            hasComputedStyle("overflow-x", isScrollOrAuto, device),
+            hasComputedStyle("overflow-y", isScrollOrAuto, device),
+          ),
+        ),
+      ),
```

Comment:
That's a good point. There is an implicit assumption that the scroll position is the default and has not been scrolled. Though I'm not sure I understand why that means checking top and left position, wont content to the left or above always be clipped? I'll leave a note in any case.

Reply:
Yes: go to page, scroll container, run extension: get flagged because something scrolled **above** the container is "too close" to the top menu.
(or maybe I'm missing a point and these are already considered invisible 🤔 )

Reply:
It seems content to the left or above, regardless of whether or not it can be scrolled (back) into view or is clipped, is not considered visible. This is probably not the behavior we want so I'll log an issue for it.
For this predicate we would want to only regard content that can be scrolled (back) into view, and not clipped, as scrolled behind. If have added a note to the jsdoc and I'll also log an issue for this.